podman_tutorial.md

```diff
@@ -0,0 +1,156 @@
+
+```
+# sudo apt-get update
+# sudo apt-get install libdevmapper-dev libglib2.0-dev libgpgme11-dev golang libseccomp-dev \
+                        go-md2man libprotobuf-dev libprotobuf-c0-dev libseccomp-dev
+# mkdir -p ~/src/github.com/projectatomic/
+# cd ~/src/github.com/projectatomic/
+# git clone https://github.com/projectatomic/libpod/
+#
+# cd libpod
+# make
+# sudo make install PREFIX=/usr
+# sudo mkdir -p /etc/containers
+# sudo bash -c 'cat <<EOF > /etc/containers/registries.conf
+registries.search
+registries = ['docker.io', 'registry.fedoraproject.org']
+EOF
+'
+# sudo bash -c 'cat <<EOF > /etc/containers/policy.json
+ {
+     "default": [
+         {
+             "type": "insecureAcceptAnything"
+         }
+     ],
+     "transports":
+         {
+             "docker-daemon":
+                 {
+                     "": [{"type":"insecureAcceptAnything"}]
+                 }
+         }
+ }
+ EOF
+'
+# git clone https://github.com/containernetworking/plugins.git ~/src/github.com/containernetworking/plugins
+# cd ~/src/github.com/containernetworking/plugins
+# ./build
+# sudo mkdir -p /usr/libexec/cni
+# sudo cp bin/* /usr/libexec/cni
+# git clone https://github.com/opencontainers/runc.git ~/src/github.com/opencontainers/runc
+# GOPATH=~/ make static BUILDTAGS="seccomp selinux"
+# sudo cp runc /usr/bin/runc
+# sudo mkdir -p /usr/local/libexec
+# sudo ln -s /usr/libexec/crio /usr/local/libexec/
+```
+
+) -->
+## Familiarizing yourself with Podman
+
+### Running a sample container
+This sample container will run a very basic httpd server that serves only its index
+page.
+```
+# sudo podman run -dt -e HTTPD_VAR_RUN=/var/run/httpd -e HTTPD_MAIN_CONF_D_PATH=/etc/httpd/conf.d \
+                    -e HTTPD_MAIN_CONF_PATH=/etc/httpd/conf \
+                    -e HTTPD_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/httpd/ \
+                    registry.fedoraproject.org/f26/httpd /usr/bin/run-httpd
+```
+Because the container is being run in detached mode, represented by the *-d* in the podman run command, podman
+will print the container ID after it has run.
+
+### Listing running containers
+The Podman *ps* command is used to list creating and running containers.
+```
+# sudo podman ps
+```
+
+Note: If you add *-a* to the *ps* command, Podman will show all containers.
+
+### Executing a command in a running container
+You can use the *exec* subcommand to execute a command in a running container.  Eventually you will be able to
+obtain the IP address of the container through inspection, but that is not enabled yet.  Therefore, we will
+install *iproute* in the container.  Notice here that we use the switch **--latest** as a shortcut for the latest
+created container.  You could also use the container's ID listed during *podman ps* in the previous step or
+when you ran the container.
+```
+# sudo podman exec --latest -t dnf -y install iproute
+# sudo podman exec --latest -t ip a
+```
+
+Note the IP address of the *eth0* device.
```

Comment:
I've not stinking eth0 device, rather an ens3 device.

Reply:
changed to ethernet